jelly/src/main/java/org/kohsuke/stapler/jelly/ResourceBundle.java

```diff
@@ -153,14 +157,24 @@ protected Properties get(String key) {
         }
 
         if(in!=null) {
+            PropertyResourceBundle propertyResourceBundle;
             try {
-                try {
-                    props.load(in);
-                } finally {
-                    in.close();
-                }
+                propertyResourceBundle = new PropertyResourceBundle(in);
             } catch (IOException e) {
                 throw new Error("Failed to load "+url,e);
+            } finally {
+                try {
+                    in.close();
+                } catch (IOException ignored) {}
+            }
+
+            Enumeration<String> keys = propertyResourceBundle.getKeys();
+            // TODO Java 9+ can use 'asIterator' and get rid of below collections conversion
+            List<String> keysAsSaneType = Collections.list(keys);
+
+            for (String localKey : keysAsSaneType) {
+                String value = propertyResourceBundle.getString(localKey);
+                props.put(localKey, value);
```

Comment:
Per [the Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html):

> Because `Properties` inherits from `Hashtable`, the `put` and `putAll` methods can be applied to a `Properties` object. Their use is strongly discouraged as they allow the caller to insert entries whose keys or values are not `String`s. The `setProperty` method should be used instead.

```suggestion
                props.setProperty(localKey, value);
```